<a href="https://colab.research.google.com/github/mohansaidinesh/Machine-Learning/blob/main/Random_Forest.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [114]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import classification_report, confusion_matrix

In [115]:
#load the dataset
data=pd.read_csv("https://assets.datacamp.com/production/repositories/628/datasets/444cdbf175d5fbf564b564bd36ac21740627a834/diabetes.csv")
data.head()

pregnancies  glucose  diastolic  triceps  insulin   bmi    dpf  age  \
0            6      148         72       35        0  33.6  0.627   50   
1            1       85         66       29        0  26.6  0.351   31   
2            8      183         64        0        0  23.3  0.672   32   
3            1       89         66       23       94  28.1  0.167   21   
4            0      137         40       35      168  43.1  2.288   33   

   diabetes  
0         1  
1         0  
2         1  
3         0  
4         1

In [116]:
data.shape

(768, 9)

In [117]:
X = data.drop('diabetes', axis=1).values
y = data['diabetes'].values

In [118]:
#split the data into training and testing data
X_train, X_test, y_train, y_test=train_test_split(X,y,test_size=0.2,random_state=45)

fitting the model

In [119]:
model = RandomForestClassifier(n_estimators=15,criterion='entropy')
model.fit(X_train, y_train)

RandomForestClassifier(criterion='entropy', n_estimators=15)

In [120]:
#scaling the testing data
y_pred = model.predict(X_test)

In [121]:
model.score(X_train, y_train)

0.996742671009772

In [122]:
model.score(X_test, y_test)

0.7402597402597403

In [123]:
#confusion matrix
confusion_matrix(y_test, y_pred)

array([[86, 18],
       [22, 28]])

In [124]:
print(classification_report(y_test, y_pred))

              precision    recall  f1-score   support

           0       0.80      0.83      0.81       104
           1       0.61      0.56      0.58        50

    accuracy                           0.74       154
   macro avg       0.70      0.69      0.70       154
weighted avg       0.74      0.74      0.74       154



**CONCLUSION :** The accuracy of training data is 1.0 and for the testing data is 0.81